changelog.d/20221117_103859_aaschaer_recursive_filter.md

```diff
@@ -0,0 +1,4 @@
+### Enhancements
+
+* the `--filter` and `--recursive` options to `globus ls` now combine better
+  with directories that don't match the name filter being explored
```

Comment:
May I suggest this longer, but hopefully more digestible, explanation?

```suggestion
* `globus ls` has improved behavior when the `--filter` and `--recursive` options
   are used in combination

  * directory names are not matched against the filter, allowing the operation to
    traverse directories regardless of their names

  * the `--filter` is still applied to filenames in all directories traversed by
    the `ls` operation

  * directory names can be filtered out of the text output by eliminating
    lines which end in `/`

  * the behaviors of `globus ls` commands with `--recursive` or `--filter`, but not
    both, are unchanged
```

I'm happy for this to get further refined, but I think we need to explain how they behave when combined here, and it's good to note that we aren't changing how they behave when _not_ combined.

Reply:
This is significantly better, so I'm going to add the suggested commit, and then we can modify as needed